### ISSUE TITLE
Can no longer user SSH host alias to install module from private repo

### ISSUE THREAD

**Issue by gavinhungry** [pnpm/pnpm#6827]
### Code to reproduce the issue:
```
pnpm i github:my_company/private_repo
```

### Expected behavior:
`pnpm` uses the `github` SSH alias to authenticate via private key, and installs the module from `private_repo`. This was the behavior in `pnpm` 8.6.7. This syntax *does* still work for public repositories.

```
# ~/.ssh/config

Host github
  HostName        github.com
  User            git
  IdentityFile    ~/.ssh/my_ssh_private_key
```

### Actual behavior:
`pnpm` instead prompts me for my GitHub username/password.

### Additional information:

 - `node -v` prints: v20.4.0
 - Windows, macOS, or Linux?: Linux

I assume this is related to #6805.

**Comment by KSXGitHub** [comment]
So solving one bug causes another bug. This is going to be a long day.

**Comment by KSXGitHub** [comment]
I can't reproduce:

![screenshot](https://github.com/pnpm/pnpm/assets/11488886/2b187e6f-8d4e-4d83-8706-b3345b43b6cc)

**Comment by gavinhungry** [comment]
I can confirm that the above change (as tested in v8.6.9) is working as expected for me.

Thank you! :+1:

**Comment by gavinhungry** [comment]
~~I can confirm that above change (as tested in v8.6.9) works as expected for me.~~

Whoops, I was reading the changelog notes from the [unmerged PR](https://github.com/pnpm/pnpm/pull/6829/files), and not the changelog from the actual [v8.6.9 release](https://github.com/pnpm/pnpm/releases/tag/v8.6.9), the latter indicating that it only reverted the fix from #6805.

I am happy to test this against your branch, however.
